### DISCUSSION TITLE
Dynamic Rendering with dynamic functions isn't working

### DISCUSSION THREAD

**Discussion by dinesh-hills** [vercel/next.js#52409] [App Router]
nextjs version: 13.4.7

Why soft navigation with `router.push("/example")` from `router/navigation` to a server component page which uses dynamic function `cookie( )` doesn't render dynamically?

It works with `router.replace` of course it reloads the page but why not when used `router.push`?

**Comment by dinesh-hills** [comment]
I tested once again, now `router.replace` isn't working either. only if I manually refresh the page the data gets updated.

**Reply by masterbater** [reply]
Probably update to 13.4.9 latest there where some improvements made .8 and .9

If you dont want to update use router.refresh() or use revalidate etc.

**Reply by dinesh-hills** [reply]
@masterbater I will update and test it, and yes I did try setting `revalidation = 0` and `dymacic='force-dynamic'` nothing seem to do the trick.

how do you use router.refresh() in my case, it is a server component.
